types/chrome/index.d.ts

```diff
@@ -3563,13 +3564,13 @@ declare namespace chrome.fileSystemProvider {
 
     export interface EntryMetadata {
         /** True if it is a directory. */
-        isDirectory: boolean;
+        isDirectory?: boolean;
```

Comment:
> isDirectory
> boolean optional
> True if it is a directory. Must be provided if requested in options.

It seems that this is required in some cases. Should this interface be used as `Required<EntryMetadata>` where they're required?

Reply:
It's a runtime required.

Like a caller code can request to know if it's a directory then this is filled with `true|false` however the caller might request to know other metadata like `mimeType` or `size`.

Reply:
Types can do that certainly: have it `Required<>` when passed in a request parameter or `Partial<>` when returned in the event

Reply:
I think I haven't explained it well.

Those properties are never required.  It can always be `undefined`, it all depends of the runtime, if you ask for it you get it, if you don't you get it, aka it's `undefined`.

Additionally, we have to follow what the actual implementation does or guarantees, the source code for this interface definition is [here](
https://source.chromium.org/chromium/chromium/src/+/main:chrome/common/extensions/api/file_system_provider.idl;l=56;drc=4a8573cb240df29b0e4d9820303538fb28e31d84)

```
  dictionary EntryMetadata {
    boolean? isDirectory;
```

The notation with `?` means that that property can be `undefined` (or `null` in the C++).

The actual C++ implementation is [here](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ash/file_system_provider/operations/get_metadata.cc;l=37-54;drc=f0bed3ca3efee1e7b21d442fa483f681c28562a9)

See the IFs where the properties aren't set unless it goes in the IF.

Reply:
What I don't understand is why the docs say that they're required then. Are Chrome's own docs wrong?

Reply:
Oh right, I see your point.

Just to make sure, you're referring to the doc here in the `index.d.ts`, right?

Because the one from the [docs site](https://developer.chrome.com/docs/extensions/reference/fileSystemProvider/#type-EntryMetadata) and from the [source code](https://source.chromium.org/chromium/chromium/src/+/main:chrome/common/extensions/api/file_system_provider.idl;l=57;drc=4a8573cb240df29b0e4d9820303538fb28e31d84) says "must be provided if requested in `options`".

The documentation here on index.d.ts is stale, this was changed over 7 years ago:
https://codereview.chromium.org/1435743008/diff/160001/chrome/common/extensions/api/file_system_provider.idl

I've updated all the field's docs for `EntryMetadata`.